### ISSUE TITLE
Non-commutative comparisons

### ISSUE THREAD

**Issue by Sp3000** [cheddar-lang/Cheddar#21]
**Version:** 0.3.0

### Problem

```
cheddar> 1 == true
 `==` has no behavior for types `Number` and `Boolean`
cheddar> true == 1
 false
```

Similarly,

```
cheddar> 1 > true
 `>` has no behavior for types `nil` and `Boolean`
cheddar> true > 1
 DATA.Operator.get(...) is not a function
```

### Desired Behavior

At the very least, the behaviour/error should be the same for both cases.

**Comment by vihanb** [comment]
>_> suck bork. very ;_:

**Comment by vihanb** [comment]
Fixed in 52c6aeac4ab8c65a463931dc49e75f6a6f986565
